### DISCUSSION TITLE
For Prometheus HA (2 relicas), is the Thanos using double object storage?

### DISCUSSION THREAD

**Discussion by patrickdung** [thanos-io/thanos#6031] [Questions & Answers] [answered]
When using Kube-prometheus, the default number of Prometheus replicas is 2 in the Jsonnet template.
Then the Thanos sidecar is used. 
So, after about two hours, is the data uploaded to object storage is doubled when the two Prometheus instance is used?

**Comment by squat** [comment, marked as answer]
Yes, both Thanos sidecars will upload data to object storage.

**Comment by patrickdung** [comment]
OK.

Is there mechanism or plan to de-duplicate these uploaded data?
Suppose these raw data won't be updated after it is uploaded.

Edit:
Found the answer myself
https://github.com/thanos-io/thanos/blob/main/docs/components/compact.md#vertical-compactions
